pxe-on-rasppi/README.md

```diff
@@ -0,0 +1,517 @@
+interface.  Change the boot order so that Network Boot is on top of
+the list.  Then save and exit.  Now the Thinkpad should boot CoreOS as
+follows:
+
+<img src="pxe-boot.jpg" width=400 />
+
+
+## Install CoreOS
+
+Above steps allows us to boot a CoreOS on the Thinkpad from Raspberry
+Pi PXE server.  This CoreOS system mounts a in-memory `tmpfs`
+filesystem at the root mount point `/`.  However, we usually want
+CoreOS installed on to local disks, so we don't rely on the PXE server
+everytime we boot a target computer.
+
+To do so, we can make use of the cloud-config mechanism of CoreOS.
+Everytime CoreOS boots, it executes a cloud-config file.  This
+cloud-config file could be a YAML file, which will be interpreted and
+translated to some system configuration files, or, more flexiblly, a
+Shell script, which is executed directly.  In above steps, we let the
+PXE-booted in-memory CoreOS executes
+`http://192.168.1.105:8080/install-coreos.sh` after
+booting.  We can draft this script to call `coreos-install`, a
+standard Shell script distributed with CoreOS, to install CoreOS onto
+the local disk.
+
+`coreos-install` does three things:
+
+1. Download the CoreOS installation image and the corresponding
+   signature file from a HTTP server specified with the `-b` command
+   line parameter.  We will make the Raspberry Pi the HTTP server.
+
+1. Install the image to a device specified with the `-d` command line
+   flag.  The specified device need to be a *disk*-typed device.  For
+   more information, please refer to
+   [this pitfall](#coreos-suppoted-deivce-types).
+
+1. Copy the cloud-config file specified with the `-c` command line
+   parameter to be `/var/lib/coreos-install/user_data`.  Then,
+   everytime the newly installed CoreOS system boots from disk, it
+   will execute this cloud-config file.  We can use this cloud-config
+   file to install and configure systems like etcd and Kubernetes.
+   
+
+### HTTP Server
+
+To host `http://192.168.2.10:8080/install-coreos.sh` and
+the CoreOS installation image etc, we install Nginx on 192.168.2.10,
+the Raspberry Pi PXE server:
+
+```
+sudo apt-get update
+sudo apt-get install nginx
+```
+
+Nginx serves files in directory `/var/www/html`.  To test that the
+Nginx server is running:
+
+```
+cd /var/www/
+sudo chown -R pi html
+cd html
+echo World > world
+```
+
+Then access Nginx from the MacBook Pro:
+
+```
+curl http://192.168.2.10/world
+World
+```
+
+### CoreOS Installation Images
+
+Download the CoreOS installation images and corresponding signature files:
+
+```
+cd /var/www/html
+mkdir 1010.5.0
+cd 1010.5.0
+wget https://stable.release.core-os.net/amd64-usr/1010.5.0/coreos_production_image.bin.bz2
+wget https://stable.release.core-os.net/amd64-usr/1010.5.0/coreos_production_image.bin.bz2.sig
+```
+
+### `install-coreos.sh`
+
+Put the following `install-coreos.sh` file in `/var/www/html`.
+
+```
+#!/bin/sh
+mac_addr=`ifconfig eth0 | head -n 1 | awk '{print $NF;}'`
```

Comment:
I figured that the name of the first network device in CoreOS is not eth0 but enp1s0f0, and the MAC address is not at the end of the first line. Could you confirm that?